apps/snitch_core/lib/core/data/schema/stock/stock_movement.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Snitch.Data.Schema.StockMovement do
+  @moduledoc """
+  Model to track inventory movement in location
+  """
+  use Snitch.Data.Schema
+  use Snitch.Data.Schema.Stock
+
+  @type t :: %__MODULE__{}
+
+  schema "snitch_stock_movements" do
+    field(:quantity, :integer, default: 0)
+    field(:action, :string)
+    field(:originator_type, :string)
```

Comment:
What is an `originator`? Do you mean `origin`? Please add a `@typedoc`.
These fields can never be set because your `cast` will throw them out.

Reply:
will be added when originator code is added. Originator can be stock location. or any other schema. But not sure. Haven't started with logics yet